homeassistant/components/sensor/synologydsm.py

```diff
@@ -226,17 +229,12 @@ class SynoNasStorageSensor(SynoNasSensor):
     @property
     def state(self):
         """Return the state of the sensor."""
-        temp_sensors = ['volume_disk_temp_avg', 'volume_disk_temp_max',
-                        'disk_temp']
-
         if self.monitor_device is not None:
-            if self.var_id in temp_sensors:
-                attr = getattr(
-                    self._api.storage, self.var_id)(self.monitor_device)
-
-                if self._api.temp_unit == TEMP_CELSIUS:
-                    return attr
+            attr = getattr(self._api.storage, self.var_id)(self.monitor_device)
 
-                return round(attr * 1.8 + 32.0, 1)
+            if attr is not None:
+                if (self.var_id in TEMP_SENSORS and
+                        self._api.temp_unit == TEMP_FAHRENHEIT):
+                    return round(attr * 1.8 + 32.0, 1)
```

Comment:
We should not do this at all. Home Assistant will automatically convert if we report in Celsius and the user preference is Fahrenheit. So instead we should set `unit_of_measurement` correctly.